pkg/datasource/csvsource/bybit_tick_downloader.go

```diff
@@ -0,0 +1,101 @@
+package csvsource
+
+import (
+	"bytes"
+	"compress/gzip"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+)
+
+func Download(symbol string, start time.Time) {
+	for {
+		var (
+			path     = fmt.Sprintf("pkg/datasource/csv/testdata/bybit/%s/", symbol)
+			fileName = fmt.Sprintf("%s%s.csv", strings.ToUpper(symbol), start.Format("2006-01-02"))
+		)
+
+		if fileExists(path + fileName) {
+			start = start.AddDate(0, 0, 1)
+			continue
+		}
+
+		var url = fmt.Sprintf("https://public.bybit.com/trading/%s/%s.gz",
+			strings.ToUpper(symbol),
+			fileName)
+
+		fmt.Println("fetching ", url)
+
+		err := readCSVFromUrl(url, path, fileName)
+		if err != nil {
+			fmt.Println(err)
+			break
+		}
+
+		start = start.AddDate(0, 0, 1)
+	}
+}
+
+func readCSVFromUrl(url, path, fileName string) error {
```

Comment:
looks like this function involves read csv from url and download to a file, could we split this function to:

content, err := readCsvFromUrl 

and 

err := writeCsvFile(content, path)

Reply:
that's much better indeed